File: square/src/main/java/com/raxdenstudios/square/utils/ResourceUtils.java
package com.raxdenstudios.square.utils;

import android.content.Context;

/**
 * Created by Ángel Gómez on 09/06/2017.
 */

public class ResourceUtils {

    public static int getLayoutId(Context context, String name) {
        int resID = 0;
        if (name != null && name.length() > 0) {
            resID = context.getResources().getIdentifier(name.replaceAll("R.layout.", ""), "layout", context.getPackageName());
        }
        return resID;
    }

}

File: square/src/main/java/com/raxdenstudios/square/utils/ResourceUtils.kt
package com.raxdenstudios.square.utils

import android.content.Context

/**
 * Created by Ángel Gómez on 09/06/2017.
 */

object ResourceUtils {

    fun getLayoutId(context: Context, name: String?): Int = name?.takeIf { it.isNotEmpty() }?.let {
        context.resources.getIdentifier(it.replace("R.layout.", ""), "layout", context.packageName)
    } ?: 0

}
